core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -618,7 +620,9 @@ object Battle {
         }
 
         // Remove improvements
-        tile.improvement = null
+        if (tile.improvement != null && !attacker.getCivInfo().gameInfo.ruleSet.tileImprovements[tile.improvement]!!.hasUnique("Indestructible")) {
```

Comment:
I'm prettu sure there's a tile.getImprovement() function or whatnot
tileInfo should have a ruleSet reference as well iirc